command/agent/agent.go

```diff
@@ -867,6 +867,16 @@ func (a *Agent) ShouldReload(newConfig *Config) (agent, http, rpc bool) {
 	a.configLock.Lock()
```

Comment:
The comment on the method is misleading, seems like this determines whether to reload the config for all cases, not just comparing TLS. Also, wouldn't this have to support diffing pretty much anything in the config that changed?